site-src/geps/gep-726.md

```diff
@@ -0,0 +1,274 @@
+In the following example, a request to `/foo/abc` would be redirected to
+`/bar/abc`.
+
+```yaml
+kind: HTTPRoute
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: http-filter-1
+spec:
+  rules:
+    - matches:
+      - path:
+          type: Prefix
+          value: /foo
+      filters:
+      - type: RequestRedirect
+        requestRedirect:
+          hostname: foo.com
+          path:
+            type: PathMatch
+            value: /bar
+```
+
+This would be represented with the following API additions:
+```go
+// HTTPRequestRedirect defines a filter that redirects a request. At most one of
+// these filters may be used on a Route rule. This may not be used on the same
+// Route rule as a HTTPRequestRewrite filter.
+type HTTPRequestRedirect struct {
+  // Path defines a path redirect.
+  //
+  // Support: Extended
+  //
+  // +optional
+  Path *HTTPPathRedirect `json:"path,omitempty"`
+  // ...
+}
+
+// HTTPPathRedirectType defines the type of path redirect.
+type HTTPPathRedirectType string
+
+const (
+  // This type of redirect indicates that the complete path will be replaced by
+  // the path redirect value.
+  AbsoluteHTTPPathRedirect HTTPPathRedirectType = "Absolute"
+
+  // This type of redirect indicates that the path match will be replaced by
+  // the path redirect value.
+  PathMatchHTTPPathRedirect HTTPPathRedirectType = "PathMatch"
+)
+
+// HTTPPathRedirect defines configuration for path redirects.
+type HTTPPathRedirect struct {
+  // Type defines the type of path redirect.
+  //
+  // +kubebuilder:validation:Enum=Absolute;PathMatch
+  Type HTTPPathRedirectType `json:"type"`
+
+  // Value defines the HTTP path value to redirect to. An empty value ("")
+  // indicates that the portion of the path to be changed in the redirect should
+  // be removed from the redirect path. For example, a request to "/foo/bar"
+  // with a prefix match of "/foo" would be redirected to "/bar".
+  //
+  // +kubebuilder:validation:MaxLength=1024
+  Value string `json:"value"`
+}
+```
+
+### Rewrites
+
+A new `RequestRewrite` filter can be added to support rewrites. In the following
+example, a a request to `example.com/foo/abc` would be rewritten to
+`example.net/bar/abc`.
+
+```yaml
+kind: HTTPRoute
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: http-filter-1
+spec:
+  hostnames:
+  - example.com
+  rules:
+    - matches:
+      - path:
+          type: Prefix
+          value: /foo
+      filters:
+      - type: RequestRewrite
+        requestRewrite:
+          hostname: example.net
+          path:
+            type: ReplacePathMatch
+            value: /bar
+```
+
+This would be represent with the following API additions:
+```go
+// HTTPRequestRewrite defines a filter that modifies a request during
+// forwarding. At most one of these filters may be used on a Route rule. This
+// may not be used on the same Route rule as a HTTPRequestRedirect filter.
+type HTTPRequestRewrite struct {
```

Comment:
What is the conformance level of this filter?

Reply:
Apparently I defined conformance at a field level instead of for the filter as a whole, but since both fields have extended conformance, I should probably move that up a level.